### PR TITLE
Add FXIOS-8988 [Microsurvey] Redux architecture

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8988)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19841)

## :bulb: Description
Setup Redux architecture to handle state for microsurvey prompt. Draft PR to discuss with Orla in how we envision our architecture to be going forward and to see microsurveys as an example. 

Handles
- initializing / showing prompt
- dismissing prompt when tapping on close button
- print out text to open survey when tapping on continue button

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Fakespot/FakespotState.swift [hunk 1]
```diff
@@ -33,17 +33,6 @@ struct FakespotState: ScreenState, Equatable {
...
-        self.init(
-            windowUUID: appState.windowUUID,
-            isOpen: appState.fakespotState.isOpen,
-            sidebarOpenForiPadLandscape: appState.fakespotState.sidebarOpenForiPadLandscape,
-            currentTabUUID: appState.fakespotState.currentTabUUID,
-            expandState: appState.fakespotState.expandState,
-            telemetryState: appState.fakespotState.telemetryState
-        )
-    }
-
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Fakespot/FakespotState.swift:46]: I am able to build without this, seems like we can remove

**Review by OrlaM — APPROVED** [review]

File: firefox-ios/Client.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3818,9 +3821,6 @@
 		436F42FD2A52E6E100E4C2A2 /* uk */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = uk; path = uk.lproj/SelectCreditCard.strings; sourceTree = "<group>"; };
 		436F68302AE68E3700BF46C9 /* nb */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = nb; path = nb.lproj/Share.strings; sourceTree = "<group>"; };
 		436F84692A4070C800279DAE /* ro */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ro; path = ro.lproj/Onboarding.strings; sourceTree = "<group>"; };
-		436FC4842BD6801200CED3C5 /* tt */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = tt; path = tt.lproj/BottomSheet.strings; sourceTree = "<group>"; };
-		436FC4852BD6801200CED3C5 /* tt */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = tt; path = tt.lproj/ContextualHints.strings; sourceTree = "<group>"; };
-		436FC4862BD6801200CED3C5 /* tt */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = tt; path = tt.lproj/Credentials.strings; sourceTree = "<group>"; };
```
[the remark this answers is not in the document]
Comment (cyndichin) [reply firefox-ios/Client.xcodeproj/project.pbxproj]: discarded these changes

File: firefox-ios/Client/Frontend/Microsurvey/Redux/MicrosurveyState.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            )
+        case MicrosurveyMiddlewareActionType.openSurvey:
+            return MicrosurveyState(
+                windowUUID: state.windowUUID,
+                showPrompt: state.showPrompt,
+                showSurvey: true,
+                model: state.model
+            )
+        default:
+            return state
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Microsurvey/Redux/MicrosurveyState.swift:60]: return `MicrosurveyState()` instead state

File: firefox-ios/nimbus-features/microsurveyFeature.yaml [hunk 4]
```diff
@@ -16,4 +16,4 @@ features:
           enabled: false
       - channel: developer
         value:
-          enabled: false
+          enabled: true
```
Comment (cyndichin) [inline firefox-ios/nimbus-features/microsurveyFeature.yaml:19]: this prompt can now be dismissed, so I will turn this on in dev mode.

**Review by OrlaM — APPROVED** [review]

**Pull request merged**